doc/release/0.17.0-notes.rst

```diff
@@ -29,6 +29,9 @@ Deprecated features
 Backwards incompatible changes
 ==============================
 
+The functions ``trim1`` and ``trimboth`` now sort elements before trimming.
+This is backwards incompatible but it made no sense to trim unsorted input.
```

Comment:
```np.partition``` will scale better, but as long as we support NumPy 1.6 and 1.7 we need to have a fallback method for those.